Objects/listobject.c

```diff
@@ -3443,19 +3443,26 @@ static PyObject *
 listiter_reduce_general(void *_it, int forward)
 {
     PyObject *list;
+    PyObject *iter;
```

Comment:
You can just declare these within the block where they are used.

Reply:
Moved the declarations to the if blocks in [049a8dd](https://github.com/python/cpython/pull/101769/commits/049a8dda075a4b44cc7bd6997b2c42b70fead05e)